src/main/java/com/github/jasminb/jsonapi/Link.java

```diff
@@ -79,4 +86,24 @@ public void setMeta(Map<String, ?> meta) {
 	public String toString() {
 		return String.valueOf(getHref());
 	}
+
+
+	protected static class LinkSerializer extends StdSerializer<Link> {
+
+		public LinkSerializer() {
+			super(Link.class);
+		}
+
+		@Override
+		public void serialize(Link link, JsonGenerator json, SerializerProvider provider) throws IOException {
+			if (link.getMeta() != null) {
```

Comment:
I'm assuming `link` cannot be null?

Reply:
Correct, from the Jackson JsonSerializer javadoc:
`value` - Value to serialize; can **not** be null.